components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/utils/ToolkitLibExtractionUtils.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.wso2.apimgt.gateway.cli.utils;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.wso2.apimgt.gateway.cli.cmd.InitCmd;
+import org.wso2.apimgt.gateway.cli.constants.GatewayCliConstants;
+import org.wso2.apimgt.gateway.cli.exception.CLIInternalException;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+/**
+ * This class represents the utility functions required for library packages extraction
+ */
+public class ToolkitLibExtractionUtils {
+    private static final Logger LOGGER = LoggerFactory.getLogger(InitCmd.class);
+
+    /**
+     * Extracts the platform and runtime and copy related jars and balos to extracted runtime and platform.
+     */
+    public static void extractPlatformAndRuntime() {
+        try {
+            String libPath = GatewayCmdUtils.getCLILibPath();
+            String baloPath = GatewayCliConstants.CLI_GATEWAY + File.separator + GatewayCliConstants.CLI_BALO;
+            String breLibPath = GatewayCliConstants.CLI_BRE + File.separator + GatewayCliConstants.CLI_LIB;
+            String runtimeExtractedPath = libPath + File.separator + GatewayCliConstants.CLI_RUNTIME;
+            String platformExtractedPath =
+                    GatewayCmdUtils.getCLILibPath() + File.separator + GatewayCliConstants.CLI_PLATFORM;
+
+            extractBallerinaDist(platformExtractedPath, libPath, baloPath, breLibPath);
+            extractBallerinaDist(runtimeExtractedPath, libPath, baloPath, breLibPath);
+
+        } catch (IOException e) {
+            String message = "Error while unzipping platform and runtime while project setup";
+            LOGGER.error(message, e);
```

Comment:
If you catch it and log it and rethrow it, then there's no way for the upstream code to know that you've already logged the exception, and so the same exception might get logged twice.

Reply:
Removed the logger message.

Reply:
Please check before whether upstream method logs it. Its better to log it twice rather than not logging it

Reply:
Whenever CLIInternalException or CLIRuntimeException is thrown, it is logged in the Main Class. So it would be better to remove that additional log message.

Reply:
I checked the code. Upstream(Main class) logs the error message.